dicoogle/src/main/java/pt/ua/dicoogle/server/users/UserExportPresets.java

```diff
@@ -43,18 +45,15 @@
      * @throws IOException if an I/O error occurs
      */
     public static void savePreset(String username, String presetName, String[] fields) throws IOException {
-        File presetsDir = new File(Platform.homePath() + "users/" + username + "/presets");
-
-        // create presets dir, if it doesn't exist
-        Files.createDirectories(presetsDir.toPath());
+        Path presetsDir = Paths.get(Platform.homePath(), "userdata", username, "presets");
 
-        File presetFile = new File(presetsDir + "/" + presetName + ".txt");
+        Files.createDirectories(presetsDir);
 
-        try (Writer writer = new PrintWriter(presetFile)) {
-            for (String field : fields) {
-                writer.write(field);
-                writer.write('\n');
-            }
+        try (ObjectOutputStream out =
+                     new ObjectOutputStream(Files.newOutputStream(Paths.get(presetsDir.toString(), presetName)))) {
```

Comment:
You can use [`Path#resolve`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#resolve-java.lang.String-) instead.

```suggestion
                     new ObjectOutputStream(Files.newOutputStream(presetsDir.resolve(presetName)))) {
```